tests/unit-tests/dds/DCPS/DispatchService.cpp

```diff
@@ -249,6 +255,43 @@ TEST(dds_DCPS_DispatchService, RecursiveDispatchDelta_IS)
   EXPECT_GE(test_obj.call_count(), 100000u);
 }
 
+TEST(dds_DCPS_DispatchService, InternalShutdown)
+{
+  OpenDDS::DCPS::DispatchService dispatcher;
+  InternalShutdownTestObj test_obj(dispatcher);
+
+  OpenDDS::DCPS::LogRestore restore;
+  OpenDDS::DCPS::log_level.set(OpenDDS::DCPS::LogLevel::None);
+
+  dispatcher.dispatch(test_obj);
+  OpenDDS::DCPS::DispatchService::EventQueue temp;
+  dispatcher.shutdown(true, &temp);
```

Comment:
I don't think so. It's mostly to check that it doesn't hang when shutdown is called internally (though it will try to complain in the log). So finishing should be enough to make sure it didn't hang.